### ISSUE TITLE
viewindex: use jsonpb to marshal compilation details [phab:D1745]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2180]
_Created by **schroederc** at 2017-06-28 17:24:25_:

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2017-06-28 17:27:42_:

this works now? woo!

**Comment by kythe** [comment]
_**shahms** wrote at 2017-06-28 17:27:42_:

I tried doing this the other day and ran into issues with unrecognized types in the Details field.  Given the internal usage, this is likely to impact quite a few CUs.

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-06-28 17:28:59_:

>>! In (earlier comment), shahms wrote:
> I tried doing this the other day and ran into issues with unrecognized types in the Details field.  Given the internal usage, this is likely to impact quite a few CUs.

We don't use this tool internally, do we?

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-06-28 17:30:39_:

Make a package-level one?

I suggest adding it to the var block above:

```
var (
   printFiles = flag.Bool(…)

   m = jsonpb.Marshaler{OrigName: true}
)
```

I think we want OrigName whether or not you move it—the camelCase names it generates don't match up with the rest of our scripts AFAIK.

**Comment by kythe** [comment]
_**fromberger** wrote at 2017-06-28 17:31:27_:

>>! In (earlier comment), schroederc wrote:
>>>! In (earlier comment), shahms wrote:
>> I tried doing this the other day and ran into issues with unrecognized types in the Details field.  Given the internal usage, this is likely to impact quite a few CUs.
> 
> We don't use this tool internally, do we?

We do, for .kindex files. But the imports should resolve the cases we care about.

**Comment by kythe** [comment]
_**shahms** wrote at 2017-06-28 17:33:05_:

THanks!

**Comment by kythe** [comment]
_**shahms** accepted this revision at 2017-06-28 17:33:05_

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-06-28 17:35:25_:

OrigName

**Comment by kythe** [comment]
_**schroederc** closed via commit a4eb648b71c3f1abe7aaa71d22651b1ee1556a1e at 2017-06-28 17:35:50_
